include/mbgl/util/geometry_buffer.hpp

```diff
@@ -0,0 +1,287 @@
+                                    EndStrategy(pointsPerCircle),
+                                    CircleStrategy(pointsPerCircle));
+
+            assert(polygons.size() > 0);
+            return lngLatPolygonFromMercator(polygons);
+        },
+        [&distanceInMeter, &bufferDistance, &pointsPerCircle](const mapbox::geometry::line_string<double>& line) {
+            // Convert meter to boost geometry distance
+            assert(line.size());
+            bufferDistance = meterInMercatorCoordinateUnits(line.front().y) * distanceInMeter;
+
+            BoostLineString boostLine;
+            boostLine.reserve(line.size());
+            for (const auto& p : line) {
+                boostLine.push_back(mercatorPointFromLngLat(p));
+            }
+            BoostMultiPolygon polygons;
+            boost::geometry::buffer(boostLine,
+                                    polygons,
+                                    DistanceStrategy(bufferDistance),
+                                    SideStrategy(),
+                                    JoinStrategy(pointsPerCircle),
+                                    EndStrategy(pointsPerCircle),
+                                    CircleStrategy(pointsPerCircle));
+
+            assert(polygons.size() > 0);
+            return lngLatPolygonFromMercator(polygons);
+        },
+        [&distanceInMeter, &bufferDistance, &pointsPerCircle](
+            const mapbox::geometry::multi_line_string<double>& lines) {
+            // Convert meter to boost geometry distance
+            assert(lines.size());
+            bufferDistance = meterInMercatorCoordinateUnits(lines.front().front().y) * distanceInMeter;
+
+            BoostMultiLineString boostLines;
+            boostLines.reserve(lines.size());
+            for (const auto& line : lines) {
+                BoostLineString boostLine;
+                boostLine.reserve(line.size());
+                for (const auto& p : line) {
+                    boostLine.push_back(mercatorPointFromLngLat(p));
+                }
+                boostLines.push_back(boostLine);
+            }
+
+            BoostMultiPolygon polygons;
+            boost::geometry::buffer(boostLines,
+                                    polygons,
+                                    DistanceStrategy(bufferDistance),
+                                    SideStrategy(),
+                                    JoinStrategy(pointsPerCircle),
+                                    EndStrategy(pointsPerCircle),
+                                    CircleStrategy(pointsPerCircle));
+
+            assert(polygons.size() > 0);
+            return lngLatPolygonFromMercator(polygons);
+        },
+        [](const auto&) {
+            mbgl::Log::Warning(mbgl::Event::General, "GeometryBuffer only supports Point/LineString geometry type.");
+            return mapbox::geometry::multi_polygon<double>();
+        });
+}
+} // namespace
```

Comment:
why all the functions above a declared in .hpp? is this header guarantied to be included only once?